src/Altinn.App.Core/Internal/Data/CachedFormDataAccessor.cs

```diff
@@ -0,0 +1,159 @@
+using System.Collections.Concurrent;
+using System.Globalization;
+using Altinn.App.Core.Features;
+using Altinn.App.Core.Internal.App;
+using Altinn.App.Core.Internal.AppModel;
+using Altinn.App.Core.Models;
+using Altinn.Platform.Storage.Interface.Models;
+
+namespace Altinn.App.Core.Internal.Data;
+
+/// <summary>
+/// Class that caches form data to avoid multiple calls to the data service for a single validation
+///
+/// Do not add this to the DI container, as it should only be created explicitly because of data leak potential.
+/// </summary>
+internal sealed class CachedInstanceDataAccessor : IInstanceDataAccessor
```

Comment:
Rename the file to match the class name?

Reply:
😊